compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TaintAnalyzer.java

```diff
@@ -204,22 +204,33 @@
     private boolean stopAnalysis;
     private boolean blockedOnWorkerInteraction;
     private BlockedNode blockedNode;
-    private Boolean taintedStatus;
-    private Boolean returnTaintedStatus;
     private Set<TaintRecord.TaintError> taintErrorSet = new LinkedHashSet<>();
     private Map<BlockingNode, List<BlockedNode>> blockedNodeMap = new HashMap<>();
+    private Map<BlockingNode, List<BlockedNode>> blockedEntryPointNodeMap = new HashMap<>();
     private Map<BLangIdentifier, Boolean> workerInteractionTaintedStatusMap;
     private BLangIdentifier currWorkerIdentifier;
     private BLangIdentifier currForkIdentifier;
 
+    // Tainted status can have three valid values. "True" to denote that the analyzed expression was "tainted". "False"
+    // to indicate that the analyzed expression was "untainted". Null to denote that the result of the expression should
+    // be neglected while evaluating the tainted status. This is used to nullify the result of recursive invocations.
+    private Boolean taintedStatus;
```

Comment:
Shall we use an Enum for this purpose?

Reply:
Did not do that since it will complicate all the boolean checks done in the downstream code. Also, BSymbol has "public boolean tainted" that maintains the tainted status against symbols. Hence, the assignments will need changes in multiple locations as well. These additional conversions will all introduce some performance overhead. 

But agreed that it can make the code much more readable. Shall we do the change as a separate improvement? If we can do so, we can do a performance comparison of this code and the code with Enum to decide which approach we can keep.

Reply:
After doing some comparison, I think it's better to use enum here. Performance impact is minimum. Will update the PR with the changes.